File: C#/myLeetCode/String/541. Reverse String II.cs
// 541. Reverse String II
// https://leetcode.com/problems/reverse-string-ii/
public class Solution {
    public string ReverseStr(string s, int k) {
        char[] arr = s.ToCharArray();
        int n = arr.Length;
        
        for (int start = 0; start < n; start += 2 * k) {
            int i = start, j = Math.Min(start + k - 1, n - 1);
            while (i < j) {
                char tmp = arr[i];
                arr[i++] = arr[j];
                arr[j--] = tmp;
            }
        }
        
        return new string(arr);
    }
}
public class Program {
    public static void Main() {
        Solution solution = new Solution();

        string input = "abcdefg";
        int k = 2;
        string result = solution.ReverseStr(input, k);

        Console.WriteLine($"Input: '{input}' with k={k}");
        Console.WriteLine($"Output: '{result}'");
    }
}

File: C++/代码随想录/String/541. Reverse String II.cpp
// 541. Reverse String II
// https://leetcode.com/problems/reverse-string-ii/

#include <iostream>

std::string reverseStr(std::string str, int k) {
    int n = str.length();
    
    for (int i = 0; i < n; i += 2 * k) {
        int left = i;
        int right = std::min(i + k - 1, n - 1);
        
        while (left < right) {
            std::swap(str[left], str[right]);
            ++left;
            --right;
        }
    }
    
    return str;
}

int main() {
    std::string str = "abcdefg";
    int k = 2;
    
    std::string result = reverseStr(str, k);
    
    std::cout << result << std::endl;
    
    return 0;
}